lib/lpm.c

```diff
@@ -58,15 +64,44 @@ lpm_lookup_ipv4(struct rte_lpm *lpm, uint32_t ip)
 	ret = rte_lpm_lookup(lpm, ntohl(ip), &next_hop);
 	if (ret == -EINVAL) {
 		G_LOG(ERR, "lpm: incorrect arguments for IPv4 lookup\n");
-		goto out;
+		return ret;
 	} else if (ret == -ENOENT) {
-		G_LOG(WARNING, "lpm: IPv4 lookup miss\n");
-		goto out;
+		/*
+		 * Failing to find an LPM entry can mean many different
+		 * things, depending on the caller. In some cases,
+		 * failing to find an entry is used as a validation of
+		 * the table, and is therefore not even an error.
+		 *
+		 * Failing to find an entry can also occur frequently
+		 * on the hot path when under attack.
+		 *
+		 * For these reasons, we are careful about how we log the
+		 * fact that the lookup failed to find an entry. We use the
+		 * DEBUG level and guess whether the entry will actually be
+		 * logged. Callers are encouraged to check for -ENOENT and
+		 * make their own log entries as needed.
+		 */
+		char buf[INET_ADDRSTRLEN];
+
+		if (!G_LOG_CHECK(DEBUG)) {
+			G_LOG(DEBUG, "lpm: IPv4 lookup miss\n");
+			return ret;
+		}
+
+		if (likely(inet_ntop(AF_INET, &ip,
+				buf, sizeof(buf)) != NULL)) {
+			G_LOG(DEBUG,
+				"lpm: IPv4 lookup miss for %s\n", buf);
+			return ret;
+		}
+
+		G_LOG(DEBUG,
+			"lpm: IPv4 lookup miss; can't convert IP to string: %s\n",
+			strerror(errno));
+		return ret;
 	}
 
 	ret = next_hop;
```

Comment:
Before this line, add `RTE_VERIFY(ret == 0);` for safety.

The same for the IPv6 case.